.github/workflows/code-coverage.yml

```diff
@@ -0,0 +1,57 @@
+name: code-coverage
+
+on:
+  push:
+    branches: [master]
+
+env:
+  java: 8
+  distribution: "adopt"
+
+jobs:
+  code-coverage:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up JDK (${{ env.java }} ${{ env.distribution }})
+        uses: actions/setup-java@v2
+        with:
+          distribution: ${{ env.distribution }}
+          java-version: ${{ env.java }}
+
+      - name: Cache Maven Packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+
+      - name: Build with Maven
```

Comment:
Is there a way to generate the coverage report without running the tests? Or maybe using an artifact from the test run done in the other workflow?

Reply:
Generally GH actions work atomically from what I've seen